From da5617f4f5d1c4a64e0fd0646eb75cbf59ea539b Mon Sep 17 00:00:00 2001
Message-Id: <da5617f4f5d1c4a64e0fd0646eb75cbf59ea539b.1380730269.git.jdenemar@redhat.com>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Mon, 30 Sep 2013 20:14:54 +0200
Subject: [PATCH] docs: Discourage users to set hard_limit

https://bugzilla.redhat.com/show_bug.cgi?id=1013758

In one of my previous patches I am removing the hard_limit heuristic to
guess the correct value if none set. However, it turned out, this limit
is hard to guess even for users. We should advise them to not set the
limit as their domains may be OOM killed. Sigh.
(cherry picked from commit 09adfdc62de2bbba71580839f735ec07a356c762)

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 docs/formatdomain.html.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/docs/formatdomain.html.in b/docs/formatdomain.html.in
index e20d1e2..c3ed5a7 100644
--- a/docs/formatdomain.html.in
+++ b/docs/formatdomain.html.in
@@ -611,7 +611,11 @@
       <dt><code>hard_limit</code></dt>
       <dd> The optional <code>hard_limit</code> element is the maximum memory
         the guest can use. The units for this value are kibibytes (i.e. blocks
-        of 1024 bytes)</dd>
+        of 1024 bytes). <strong>However, users of QEMU and KVM are strongly
+        advised not to set this limit as domain may get killed by the kernel.
+        To determine the memory needed for a process to run is
+        <a href="http://en.wikipedia.org/wiki/Undecidable_problem">
+        undecidable problem</a>.</strong></dd>
       <dt><code>soft_limit</code></dt>
       <dd> The optional <code>soft_limit</code> element is the memory limit to
         enforce during memory contention. The units for this value are
-- 
1.8.3.2

